Added an info box for Giants' Foundry

Aiadan opened the pull request:
Info box shows total amount of usable metal for Giants' Foundry, with possibility to filter for any combination of equipment, metal bars and ore.

Aiadan commented:
@riktenx have you had the opportunity to look into this?

riktenx commented:
@Aiadan Sorry, I was given collab but I didn't actually click the watch button on the repo. I've done that now. Glad you pinged me so I was notified. I will look at this today.

riktenx commented:
FYI I am fine with making the above changes myself so we can get this merged, if you would prefer that.

Aiadan commented:
> Off the cuff this should really be integrated into the existing 2d overlay with toggles to control it. Since that already only shows in Giants' foundry, the location stuff you added wouldn't be needed.

I like having the option to show it outside Giants' Foundry, one of my use cases for the summary is checking if I have enough loot to make it worth to go to Giants' Foundry to begin with.
> This would all also have to be off by default, since the plugin has over 200k installs, we don't want to push out big changes that might cause excess support traffic in the main Discord.

I changed the default to not show anywhere.
> Your inventory counting code looks fine, but the overlay should display something like "open the bank" when they have not yet done so, to reflect that you're missing information that you need them to action. Otherwise if they didn't have anything in their inventory, it just doesn't show anything which can be confusing.

Added a red "Open bank" text if you haven't opened the bank yet.
> Oh, one more thing -- you actually need to "snapshot" the counts on itemContainerChanged -- at the moment you're iterating the entire bank and re-doing the math every frame, which is liable to degrade client performance.

I updated the count to only update when itemContainerChanged triggers.

Aiadan commented:
> > I like having the option to show it outside Giants' Foundry
> 
> That makes sense, my main objection is having the new overlay. So I would be fine with you adding that foundry/everywhere toggle as long as your counters go into the existing overlay.

Is the suggestion to bake in the metal info in FoundryOverlay2D? It feels like the purpose of that box is different enough for them to be separated codewise.

riktenx commented:
Maybe to clarify-- being separate _codewise_ isn't the same as being together in the same panel. My request is only that you re-use the existing panel to display this new information, so users don't now have to contend with position of two panels for the one plugin. You do not necessarily have to cram all the new counting logic into that panel, it can be its own entity in code.
